### PR TITLE
Fix Blazor Server Reconnection Mechanism

### LINKED ISSUE #32545: There is no `Retry` button in the page after reconnection to the server for blazor server project
**Repro Steps**
1. Open VS > Create a new project > Blazor Server App > 6.0 > Create
2. F5 to run the application
3. After it pop up a browser to show the page, please open another browser that the website link same as previous one, then stop debugging
4. Wait for about 3 mins, verify the page info

Note: This issue does not repro on 5.0/3.1 project

**Actual Result**
After stop debugging, connect server message show up and then the page show "reload" message. Wait for about 3 mins, the page still show "Reload"
image

**Expected Result**
image

### PR DESCRIPTION
**Pull request by TanayParikh**
Fixes https://github.com/dotnet/aspnetcore/issues/32545

### CODE REVIEW

File: src/Components/Web.JS/src/Boot.Server.ts [hunk 1]
```diff
@@ -29,23 +29,8 @@ async function boot(userOptions?: Partial<CircuitStartOptions>): Promise<void> {
...
 
-  const initialConnection = await initializeConnection(options, logger, circuit);
-  const circuitStarted = await circuit.startCircuit(initialConnection);
-  if (!circuitStarted) {
-    logger.log(LogLevel.Error, 'Failed to start the circuit.');
-    return;
-  }
-
-  const reconnect = async (existingConnection?: HubConnection): Promise<boolean> => {
+  Blazor.reconnect = async (existingConnection?: HubConnection): Promise<boolean> => {
```
Comment (TanayParikh) [inline src/Components/Web.JS/src/Boot.Server.ts:33]: We were setting this _after_ we tried to init the `DefaultReconnectionHandler` which needs this `reconnect` function, leading to some null refs.

File: src/SignalR/clients/ts/signalr/src/Errors.ts [hunk 2]
```diff
@@ -158,6 +158,32 @@ export class FailedToStartTransportError extends Error {
     }
 }
 
+/** Error thrown when the negotiation with the server failed to complete. */
+/** @private */
+export class FailedToNegotiateWithServerError extends Error {
```
Comment (TanayParikh) [inline src/SignalR/clients/ts/signalr/src/Errors.ts:163]: cc/ @BrennanConroy @halter73 just adding this additional ErrorType, similar to what we did before for the Transport errors. Could I get a signoff for this part if it's okay with you?

**Review by javiercn — APPROVED** [review]
Looks good.

I trust you've done due diligence to manually test this, since it's hard/impossible to test it with automation.

:shipit: as long as the signalr folks are happy with the changes there.

**Comment by TanayParikh** [comment]
> :shipit: as long as the signalr folks are happy with the changes there.

cc/ @BrennanConroy @halter73

**Review by BrennanConroy — APPROVED** [review]

**Comment by TanayParikh** [comment]
@dotnet/aspnet-build could this please be merged. The helix test keeps timing out at 4 hours (tried retry). 

The commit prior to the merge commit passed as expected: 259c84feee6982c3640513a06379b4480269d6f6

**Comment by dougbu** [comment]
@TanayParikh I suspect the problem is a known issue. I retried the failed jobs because Core-Eng drained the macOS 10.14 queue.

**Comment by TanayParikh** [comment]
> @TanayParikh I suspect the problem is a known issue. I retried the failed jobs because Core-Eng drained the macOS 10.14 queue.

Ah okay. The backport PR is also facing the same issue of 4h timeouts on the helix queue. I've retriggered that as well, 🤞 it works as expected now.

**Pull request merged**